packages/pocl-job/src/staging/__tests__/pocl-validation-errors.spec.js

```diff
@@ -59,107 +62,325 @@ describe('pocl-validation-errors', () => {
+    })
+
+    it('leaves the payment method blank if payment source is not postal order', async () => {
+      const poclValidationError = getPoclValidationError({ methodOfPayment: null, newPaymentSource: { label: 'Magic Beans' } })
+      salesApi.getPoclValidationErrorsForProcessing.mockResolvedValue([poclValidationError])
+
+      await processPoclValidationErrors()
+
+      const finaliseTransactionPayload = salesApi.finaliseTransaction.mock.calls[0][1]
+      expect(finaliseTransactionPayload.payment.method).toBeUndefined()
+    })
+
+    it('leaves the payment method blank if payment source is not set', async () => {
+      const poclValidationError = getPoclValidationError({ methodOfPayment: null, newPaymentSource: null })
+      salesApi.getPoclValidationErrorsForProcessing.mockResolvedValue([poclValidationError])
+
+      await processPoclValidationErrors()
+
+      const finaliseTransactionPayload = salesApi.finaliseTransaction.mock.calls[0][1]
+      expect(finaliseTransactionPayload.payment.method).toBeUndefined()
+    })
+  })
+
+  describe('when a date is not in ISO format', () => {
+    it('converts the issueDate to ISO format without milliseconds', async () => {
+      const poclValidationError = getPoclValidationError()
+      poclValidationError.transactionDate = '01/01/2023'
+
       salesApi.getPoclValidationErrorsForProcessing.mockResolvedValue([poclValidationError])
       salesApi.createTransactions.mockResolvedValue([{ statusCode: 201, response: { id: 'test-response-id' } }])
-      salesApi.finaliseTransaction.mockRejectedValueOnce(getFinalisationError(500, 'Internal server error'))
       await processPoclValidationErrors()
+
+      const [[[createTransactionPayload]]] = salesApi.createTransactions.mock.calls
+      const expectedDate = '2023-01-01T00:00:00Z'
+      expect(createTransactionPayload.permissions[0].issueDate).toBe(expectedDate)
     })
 
-    it('creates transaction in the Sales Api', () => {
+    it('converts the startDate to ISO format without milliseconds', async () => {
+      const poclValidationError = getPoclValidationError()
+      poclValidationError.startDate = '01/01/2023'
+
+      salesApi.getPoclValidationErrorsForProcessing.mockResolvedValue([poclValidationError])
+      salesApi.createTransactions.mockResolvedValue([{ statusCode: 201, response: { id: 'test-response-id' } }])
+      await processPoclValidationErrors()
+
       const validationError = salesApi.createTransactions.mock.calls[0][0][0]
-      expect(validationError).toMatchSnapshot()
+      const expectedDate = '2023-01-01T00:00:00Z'
+      expect(validationError.permissions[0].startDate).toBe(expectedDate)
     })
 
-    it('attempts to finalise transaction in the Sales Api', () => {
-      const record = salesApi.finaliseTransaction.mock.calls[0]
-      expect(record).toMatchSnapshot()
+    it('converts the startDate to ISO format without milliseconds', async () => {
```

Comment:
Can we tweak the name of this test so it is clear how it is different from the one on line 347?